contracts/jobs/JobsManager.sol

```diff
@@ -26,40 +37,122 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
+        uint256 endSlashingBlock;                          // End of slashing period for this claim
+        uint256 transcoderTotalStake;                      // Transcoder's total stake at the time of claim
+        mapping (uint256 => bool) segmentVerifications;    // Mapping segment number => whether segment was submitted for verification
+        ClaimStatus status;                                // Status of claim (pending, slashed, complete)
+    }
+
+    // States of a transcode claim
+    enum ClaimStatus { Pending, Slashed, Complete }
+
+    // Transcode jobs
+    mapping (uint256 => Job) public jobs;
+    // Number of jobs created. Also used for sequential identifiers
+    uint256 public numJobs;
 
     // Check if sender is Verifier contract
     modifier onlyVerifier() {
-        if (msg.sender != address(verifier)) throw;
+        require(msg.sender == address(verifier));
+        _;
+    }
+
+    modifier jobExists(uint256 _jobId) {
+        require(_jobId < numJobs);
+        _;
+    }
+
+    modifier claimExists(uint256 _jobId, uint256 _claimId) {
+        require(_claimId < jobs[_jobId].claims.length);
         _;
     }
 
     // Events
     event NewJob(address indexed transcoder, address indexed broadcaster, uint256 jobId);
+    event NewClaim(address indexed transcoder, uint256 indexed jobId, uint256 claimId);
+    event ReceivedVerification(uint256 indexed jobId, uint256 indexed claimId, uint256 segmentNumber, bool result);
 
-    function JobsManager(address _verifier) {
+    function JobsManager(address _registry, address _token, address _verifier)
+        Manager(_registry)
+    {
+        // Set LivepeerToken address
+        token = LivepeerToken(_token);
         // Set Verifier address
         verifier = Verifier(_verifier);
-
+        // TODO: These are all test values and need to be replaced with realistic values
         // Verify all segments.
-        // TODO: This is a test value. We will need to provide a realistic default value
         verificationRate = 1;
-
         // A job becomes inactive 100 blocks after endJob() is called
-        jobEndingPeriod = 100;
-
+        jobEndingPeriod = 50;
         // A transcoder has 100 blocks for verification after claiming work
-        verificationPeriod = 100;
+        verificationPeriod = 50;
+        // Slashing period after verification is 100 blocks
+        slashingPeriod = 50;
+        // Stake slashed by 20% for a failed verification
+        failedVerificationSlashAmount = 20;
+        // Stake slashed by 30% for a missed verification
+        missedVerificationSlashAmount = 30;
     }
 
     /*
-     * @dev Returns BondingManager
+     * @dev Deposit funds for jobs
+     * @param _amount Amount to deposit
      */
-    function bondingManager() internal constant returns (IBondingManager) {
-        LivepeerProtocol protocol = LivepeerProtocol(controller);
+    function deposit(uint256 _amount) external returns (bool) {
```

Comment:
Any advantage to using `deposit` and `withdraw` transactions instead of just using the ERC20 `approve` method? Whatever the `allowance` is for the protocol could be considered the deposited balance at the moment. Have to dig into ERC20 to make sure `allowance` updates correctly if their balance changes though.